src/opening-period/rule/Rule.tsx

```diff
@@ -151,51 +152,50 @@ export default function Rule({
         },
       })
     ),
+    ...hardCodedFrequencyOptions,
   ];
 
-  const isKnownFrequencySelected = knownFrequencyValues.find(
-    ({ value }) =>
-      value.frequency_modifier === currentFrequency.frequency_modifier &&
-      value.frequency_ordinal === currentFrequency.frequency_ordinal
+  const isFrequencySelected = (value: Frequency): boolean =>
+    value.frequency_modifier === selectedFrequency.value.frequency_modifier &&
+    value.frequency_ordinal === selectedFrequency.value.frequency_ordinal;
+
+  const isKnownFrequencySelected = knownFrequencyValues.find(({ value }) =>
+    isFrequencySelected(value)
   );
 
   const allFrequencyValues: FrequencyOption[] = isKnownFrequencySelected
     ? knownFrequencyValues
-    : [
-        ...knownFrequencyValues,
-        {
-          label: generateUnknownFrequencyLabel(
-            currentFrequency.frequency_ordinal,
-            frequencyModifierOptions.find(
-              (modifierOption: InputOption) =>
-                modifierOption.value === currentFrequency.frequency_modifier
-            )?.label
-          ),
-          value: currentFrequency,
-        },
-      ];
+    : [...knownFrequencyValues, currentFrequencyAsOption];
 
-  const frequencyOptions: InputOption[] = allFrequencyValues.map(
-    convertFrequencyToInputOption
-  );
+  const [isStartDisabled, setIsStartDisabled] = useState<boolean>(false);
 
   const frequencyModifierField = `${ruleNamePrefix}.frequency_modifier`;
   const frequencyOrdinalField = `${ruleNamePrefix}.frequency_ordinal`;
+  const startField = `${ruleNamePrefix}.start`;
 
-  const onFrequencyChange = (selected: InputOption): void => {
-    const selectedFrequency = allFrequencyValues.find(
-      ({ label }) => label === selected.label
-    );
+  useEffect(() => {
+    const selectedModifier = selectedFrequency?.value.frequency_modifier;
 
-    setValue(
-      frequencyModifierField,
-      selectedFrequency?.value.frequency_modifier || null
-    );
+    setValue(frequencyModifierField, selectedModifier || null);
     setValue(
       frequencyOrdinalField,
       selectedFrequency?.value.frequency_ordinal || null
     );
-  };
+
+    if (selectedModifier) {
```

Comment:
Does this mean it is either 'pariton' or 'parillinen'?

Reply:
Yes, start does not apply anything to the rule when it has a modifier. I could add a constant which would make easier to read tough 🤔

Reply:
I added a comment instead :grin